test/refactoring/ConvertAsync/basic.swift

```diff
@@ -177,29 +354,81 @@ func alreadyThrows(completion: (String) -> Void) throws { }
+// RUN: %refactor-check-compiles -add-async-alternative -dump-text -source-filename %s -pos=%(line+1):1 | %FileCheck -check-prefix VOID-AND-ERROR-HANDLER %s
 func voidAndErrorCompletion(completion: (Void?, Error?) -> Void) {}
+// VOID-AND-ERROR-HANDLER: {
+// VOID-AND-ERROR-HANDLER-NEXT: detach {
+// VOID-AND-ERROR-HANDLER-NEXT: do {
+// VOID-AND-ERROR-HANDLER-NEXT: try await voidAndErrorCompletion()
+// VOID-AND-ERROR-HANDLER-NEXT: completion((), nil)
+// VOID-AND-ERROR-HANDLER-NEXT: } catch {
+// VOID-AND-ERROR-HANDLER-NEXT: completion(nil, error)
+// VOID-AND-ERROR-HANDLER-NEXT: }
+// VOID-AND-ERROR-HANDLER-NEXT: }
+// VOID-AND-ERROR-HANDLER-NEXT: }
 // VOID-AND-ERROR-HANDLER: func voidAndErrorCompletion() async throws {}
 
-// RUN: %refactor -add-async-alternative -dump-text -source-filename %s -pos=%(line+1):1 | %FileCheck -check-prefix TOO-MUCH-VOID-AND-ERROR-HANDLER %s
+// RUN: %refactor-check-compiles -add-async-alternative -dump-text -source-filename %s -pos=%(line+1):1 | %FileCheck -check-prefix TOO-MUCH-VOID-AND-ERROR-HANDLER %s
 func tooMuchVoidAndErrorCompletion(completion: (Void?, Void?, Error?) -> Void) {}
+// TOO-MUCH-VOID-AND-ERROR-HANDLER: {
+// TOO-MUCH-VOID-AND-ERROR-HANDLER-NEXT: detach {
+// TOO-MUCH-VOID-AND-ERROR-HANDLER-NEXT: do {
+// TOO-MUCH-VOID-AND-ERROR-HANDLER-NEXT: try await tooMuchVoidAndErrorCompletion()
+// TOO-MUCH-VOID-AND-ERROR-HANDLER-NEXT: completion((), (), nil)
+// TOO-MUCH-VOID-AND-ERROR-HANDLER-NEXT: } catch {
+// TOO-MUCH-VOID-AND-ERROR-HANDLER-NEXT: completion(nil, nil, error)
+// TOO-MUCH-VOID-AND-ERROR-HANDLER-NEXT: }
+// TOO-MUCH-VOID-AND-ERROR-HANDLER-NEXT: }
+// TOO-MUCH-VOID-AND-ERROR-HANDLER-NEXT: }
 // TOO-MUCH-VOID-AND-ERROR-HANDLER: func tooMuchVoidAndErrorCompletion() async throws {}
 
+// RUN: %refactor-check-compiles -add-async-alternative -dump-text -source-filename %s -pos=%(line+1):1 | %FileCheck -check-prefix VOID-PROPER-AND-ERROR-HANDLER %s
+func tooVoidProperAndErrorCompletion(completion: (Void?, String?, Error?) -> Void) {}
+// VOID-PROPER-AND-ERROR-HANDLER: {
+// VOID-PROPER-AND-ERROR-HANDLER-NEXT: detach {
+// VOID-PROPER-AND-ERROR-HANDLER-NEXT: do {
+// VOID-PROPER-AND-ERROR-HANDLER-NEXT: let result = try await tooVoidProperAndErrorCompletion()
+// VOID-PROPER-AND-ERROR-HANDLER-NEXT: completion((), result.1, nil)
+// VOID-PROPER-AND-ERROR-HANDLER-NEXT: } catch {
+// VOID-PROPER-AND-ERROR-HANDLER-NEXT: completion(nil, nil, error)
+// VOID-PROPER-AND-ERROR-HANDLER-NEXT: }
+// VOID-PROPER-AND-ERROR-HANDLER-NEXT: }
+// VOID-PROPER-AND-ERROR-HANDLER-NEXT: }
+// VOID-PROPER-AND-ERROR-HANDLER: func tooVoidProperAndErrorCompletion() async throws -> (Void, String) {}
+
 // 2. Check that the various ways to call a function (and the positions the
 //    refactoring is called from) are handled correctly
 
+class MyClass {}
+
+func simpleClassParam(completion: (MyClass) -> Void) { }
+
+// TODO: We cannot check that the refactored code compiles because 'simple' and
```

Comment:
Ah yeah, as mentioned above we could add the async variants. But this TODO is fine 👍